core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -446,19 +446,18 @@ public void close() {
   }
 
   /**
-   * Periodic log config check report.
+   * Periodically log the config check report.
    */
   private final class LogConfigReportHeartbeatExecutor implements HeartbeatExecutor {
-
-    /**
-     * Constructs a new {@link LogConfigReportHeartbeatExecutor}.
-     */
-    public LogConfigReportHeartbeatExecutor() {
-    }
+    private boolean mFirst = true;
 
     @Override
     public void heartbeat() {
-      mConfigChecker.logConfigReport();
+      // Skip the first heartbeat since it happens before servers have time to register their
```

Comment:
why don't we set `mFirst = false` after the first heartbeat?

Reply:
good catch, fixed

Reply:
Does this not apply to the second heartbeat, why not check on if the configuration have been registered?